docs/Compiling.md

```diff
@@ -8,13 +8,19 @@ GHC 8.6 and above are recommended.  For best performance use GHC 8.8 or
 8.10 along with `fusion-plugin` (see below).  Benchmarks show that GHC
 8.8 has significantly better performance than GHC 8.6 in many cases.
 
-GHC 9.0 and GHC 9.2 have some performance issues, please see [this
+GHC 9.0 and GHC 9.2.1 have some performance issues, please see [this
 issue](https://github.com/composewell/streamly/issues/1061) for details.
-However, upcoming releases may fix some of these issues.
+
+However, the fixes required in streamly for the majority of performance
+regressions have been implemented. Please see [this
+issue](https://github.com/composewell/streamly/pull/1462) for details.
+
+The performance will be on par with previous compilers once GHC fixes land in
+GHC 9.2.2.
```

Comment:
We can keep it terse.

"GHC 9.0 and GHC 9.2.1 have some performance issues, please see [this
issue](https://github.com/composewell/streamly/issues/1061) for details. 
GHC 9.2.2 is expected to have the fixes that will bring the performance on par with previous versions."